### ISSUE TITLE
Smart Contract Wallet

### ISSUE THREAD

**Issue by DAC-Philantrop** [AlphaWallet/alpha-wallet-ios#5512]
Restructuring Alpha Wallet to make it useable as a Smart Contract Wallet.
---
I like the all encompassing functionality offered by Alpha Wallet and I want to change the implementation to use a Smart Contract wallet to also enable social recovery and other call things like relaying.

I would appreciate any comments and feedback on the idea and it´s doability.

**Comment by hboon** [comment]
@DAC-Philantrop thanks for the interest. Would you help verify which one you are referring to?

A. to submitting changes to upstream to make AlphaWallet switch to only use smart contract wallets
B. to submitting changes to upstream to make AlphaWallet to support smart contract wallets on top of HD wallets
C. to fork AlphaWallet to switch to only use smart contract wallets
D. to fork AlphaWallet to support smart contract wallets on top of HD wallets

**Comment by DAC-Philantrop** [comment]
Thanks for the reply ! 
I want to do C, as it seems to me it is by far the least complex.

**Comment by hboon** [comment]
Yes, C is doable but not simple. You'll likely have some server infrastructure and at least some changes to the app to include  functionality such as wallet recovery and overall support for funding the wallets.
